src/cfml/system/endpoints/HTTP.cfc

```diff
@@ -81,7 +81,7 @@ component accessors=true implements="IEndpoint" singleton {
 
 		// Validate the binary hash
 		if( len( binaryHash ) && binaryHash != hash( fileReadBinary( fullPath ), "MD5" ) ) {
-			throw( 'The binary hash of the downloaded file does not match the expected hash.', 'endpointException' );
+			throw( 'The hash of the downloaded file [#hash( fileReadBinary( fullPath ))#] doesn''t match the excepted hash [#binaryHash#] #fullPath#', 'endpointException' );
```

Comment:
Please put the hash in a variable instead of calculating it again.  For a large file, this may be a non-trivial operation.

Reply:
done